src/main/java/com/google/api/generator/gapic/composer/ServiceClientClassComposer.java

```diff
@@ -622,16 +629,25 @@ private static MethodDefinition createLroAsyncMethod(
 
   private static MethodDefinition createLroCallable(
       String serviceName, Method method, Map<String, TypeNode> types) {
-    return createCallableMethod(serviceName, method, types, true);
+    return createCallableMethod(serviceName, method, types, true, false);
   }
 
   private static MethodDefinition createCallableMethod(
       String serviceName, Method method, Map<String, TypeNode> types) {
-    return createCallableMethod(serviceName, method, types, false);
+    return createCallableMethod(serviceName, method, types, false, false);
+  }
+
+  private static MethodDefinition createPagedCallableMethod(
+      String serviceName, Method method, Map<String, TypeNode> types) {
+    return createCallableMethod(serviceName, method, types, false, true);
   }
 
   private static MethodDefinition createCallableMethod(
-      String serviceName, Method method, Map<String, TypeNode> types, boolean isLroCallable) {
+      String serviceName,
+      Method method,
+      Map<String, TypeNode> types,
+      boolean isLroCallable,
+      boolean isPaged) {
```

Comment:
Instead of adding a boolean, can we have this take an enum?

`enum CallableMethodKind { REGULAR, LRO, PAGED }`